### PR TITLE
Add the cluster registry API design to the community repo

### PR DESCRIPTION
**Pull request by perotinus**
/assign @quinton-hoole @csbell 

/cc @pmorie @font @ericchiang 

This is a mechanical conversion of the [Cluster Registry API doc](https://docs.google.com/document/d/1Oi9EO3Jwtp69obakl-9YpLkP764GZzsz95XJlX1a960/edit), with a few tweaks to better suit the format and to note the provenance of this document.

Quinton and Christian, I'd like both of you to sign-off on this before I commit. Anyone else is free to chime in if they have concerns, but I believe that we achieved consensus on this doc, and I'd like to formalize that.

### CODE REVIEW

File: contributors/design-proposals/multicluster/cluster-registry/api-design.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+-   Policy-based and jurisdictional placement of workloads
+
+## API
+
+This document defines the cluster registry API. It is an evolution of the
+[current Federation cluster API](https://github.com/kubernetes/kubernetes/blob/master/federation/apis/federation/types.go#L99),
+and is designed more specifically for the "cluster registry" use case in
+contrast to the Federation `Cluster` object, which was made for the
+active-control-plane Federation. All the names in the API are tentative and
+subject to change if we find better alternatives to describe those entities.
```
Comment (csbell) [inline contributors/design-proposals/multicluster/cluster-registry/api-design.md]: I would remove this last sentence as part of this version of the doc. We already say below that the API itself is kept elsewhere.

Comment (perotinus) [reply contributors/design-proposals/multicluster/cluster-registry/api-design.md]: Good point, done.

**Comment by perotinus** [comment]
Thanks for the review, @csbell!

**Comment by perotinus** [comment]
@quinton-hoole ping?

**Comment by perotinus** [comment]
/cc @quinton-hoole 
I'm planning to submit this this evening, unless I hear otherwise from you.

**Pull request merged**